ironfish/src/utils/bech32m.ts

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { bech32m } from 'bech32'
+
+// 1023 is maximum character count where the checksum will still catch errors
+export function bech32mDecode(bech32: string): string | null {
+  try {
+    const decodedOutput = bech32m.decode(bech32, 1023)
+    const decodedWords = decodedOutput.words
+    const decodedBytes = bech32m.fromWords(decodedWords)
+    return Buffer.from(decodedBytes).toString()
+  } catch (e) {
+    return null
```

Comment:
You don't wan't to catch errors like this in JS, because errors are untyped. It won't surface decoding errors properly.